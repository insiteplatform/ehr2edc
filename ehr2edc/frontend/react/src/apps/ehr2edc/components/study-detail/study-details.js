import React from "react"
import {Link, Route, Switch} from "react-router-dom";
import {ehr2edcRoutes as routes} from "../../ehr2edc.routes";
import {
    SearchResultsError,
    SearchResultsLoading
} from "../../../../common/components/search-result/search-result.helper"
import {BreadcrumbLink} from "../../../../common/components/router/routing-breadcrumb";
import RequestError from "../../../../common/components/error/backend-request-error";
import Notification from "../../../../common/components/notification";
import {connect} from "react-redux";
import NavigationSidebar from "../navigation-sidebar/navigation-sidebar";
import api from "../../../../api/api";
import {
    eventDefinitionsApi,
    investigatorApi,
    patientIdentifierSourcesApi,
    patientRegistrationApiForStudyDetailsComponent,
    subjectObservationSummaryApi
} from "../../../../api/api-ext";

export class StudyDetails extends React.Component {

    constructor(props) {
        super(props);
        this.notifications = React.createRef();

        //const subjectStudyApi = subjectApi("/ehr2edc/study/${studyId}/subjects/${subjectId}");
        this.state = {
            studyPageUrl: props.hasOwnProperty("studyPageUrl") ? props.studyPageUrl : "/",
            errorMessage: props.hasOwnProperty("errorMessage") ? props.errorMessage : "No study data found",
            studyApi: api.posts("/ehr2edc/studies"),
            eventApi: eventDefinitionsApi(props.studyId),
            potentialInvestigatorApi: api.posts("/ehr2edc/users?status=unassigned&studyId=" + props.studyId),
            patientRegistrationApi: patientRegistrationApiForStudyDetailsComponent("/ehr2edc/studies/" + props.studyId + "/subjects"),

            investigatorApi: investigatorApi(props.studyId),
            patientIdentifierSourcesApi: patientIdentifierSourcesApi(props.studyId),
            subjectObservationSummaryApi: subjectObservationSummaryApi(props.studyId),
            potentialInvestigators: [],
            isStudyLoading: true,
            isPotentialInvestigatorsLoading: true,
            isPatientIdentifierSourcesLoading: true,
            inError: false,
            study: {
                id: props.studyId
            },
            populateEventActionText: props.populateEventActionText,
            patientIdentifierSources: [],
            contactAddress: props.contactAddress ? props.contactAddress : '',
            breadcrumbParams: {
                studyName: '',
                edcSubjectReference: '',
                eventName: ''
            },
            onInvestigatorAdd: investigator => this.assignInvestigator(investigator),
            onInvestigatorUnassign: investigator => this.unassignInvestigator(investigator),
            onEdcSubjectReferenceChanged: edcSubjectReference => this.updateEdcSubjectReference(edcSubjectReference),
            onEventChanged: event => this.updateEventName(event),
            onRefresh: () => this.fetchStudy()
        };
    }

    componentDidMount() {
        this.fetchStudy();
        this.fetchPotentialInvestigators();
        this.fetchPatientIdentifierSources();
    }

    async fetchStudy() {
        try {
            const response = await this.state.studyApi.getOne(this.state.study);
            this.setState({
                isStudyLoading: false,
                study: response.data
            });
            this.updateStudyName(response.data.name);
        } catch (response) {
            this.setState({
                isStudyLoading: false,
                inError: true,
                error: response
            });
        }
    }

    async fetchPotentialInvestigators() {
        try {
            const response = await this.state.potentialInvestigatorApi.getAll();
            this.setState({
                isPotentialInvestigatorsLoading: false,
                potentialInvestigators: response.data.potentialInvestigators
            });
        } catch (response) {
            this.setState({
                isPotentialInvestigatorsLoading: false,
                inError: true,
                error: response,
                potentialInvestigators: []
            })
        }
    }

    async fetchPatientIdentifierSources() {
        try {
            const response = await this.state.patientIdentifierSourcesApi.getAll();
            this.setState({
                isPatientIdentifierSourcesLoading: false,
                patientIdentifierSources: response.data
            });
        } catch (response) {
            this.setState({
                isPatientIdentifierSourcesLoading: false,
                inError: true,
                patientIdentifierSources: []
            })
        }
    }

    async assignInvestigator(investigator) {
        this.setAddInvestigatorInProgress(true);
        try {
            await this.state.investigatorApi.create({investigatorId: investigator.id});
            await this.fetchStudy();
            this.fetchPotentialInvestigators();
        } catch (response) {
            this.notifications.current.show(
                <RequestError error={response} contact={this.state.contactAddress}/>
                , "warning");
        }
        this.setAddInvestigatorInProgress(false)
    }

    async unassignInvestigator(investigator) {
        this.setRemoveInvestigatorInProgress(true);
        try {
            await this.state.investigatorApi.delete(investigator);
            await this.fetchStudy();
            this.fetchPotentialInvestigators();
        } catch (response) {
            this.notifications.current.show(
                <RequestError error={response} contact={this.state.contactAddress}/>
                , "warning")
        }
        this.setRemoveInvestigatorInProgress(false);
    }

    render() {
        const {study} = this.state;
        return !!study ? <>
            <NavigationSidebar study={study} studyId={study.id} studyName={study.name}/>
            {this.renderMainPane()}
            <Notification ref={this.notifications}/>
        </> : null;
    }

    renderMainPane() {
        return <div id={"study-details-main-pane"} className="grid-block vertical">
            {this.renderBreadcrumbs()}
            <div className="grid-block" id={"study-details-content-pane"}>
                <div className="grid-block vertical">
                    {this.renderContentPane()}
                </div>
            </div>
        </div>;
    }

    renderContentPane() {
        if (this.isLoading()) {
            return (<SearchResultsLoading/>);
        }
        if (this.state.inError) {
            return (
                <SearchResultsError message={this.state.errorMessage}>
                    <RequestError error={this.state.error} contact={this.state.contactAddress}/>
                </SearchResultsError>);
        }
        return this.renderStudyDetails();
    }

    isLoading() {
        return this.state.isStudyLoading || this.state.isPotentialInvestigatorsLoading || this.state.isPatientIdentifierSourcesLoading;
    }

    renderBreadcrumbs() {
        return <ul id="page-bread-crumbs" className="bread-crumbs shrink grid-block border-bottom">
            <li><Link to={"/"}>EHR2EDC Studies</Link></li>
            {this.renderStudyNameBreadcrumb()}
            {this.renderChildBreadcrumbs()}
        </ul>;
    }

    renderStudyNameBreadcrumb() {
        return routes.filter(route => route.default)
            .map((route, index) => {
                return (<li key={index} className="breadcrumb-study">
                    <Link to={route.path}>
                        {this.state.study.name}
                    </Link>
                </li>)
            });
    }

    renderChildBreadcrumbs() {
        return routes
            .sort((a, b) => a.navigation.order - b.navigation.order)
            .map((route, index) => {
                return (<BreadcrumbLink key={index}
                                        path={route.path}
                                        default={route.default}
                                        title={route.navigation.title}
                                        breadcrumbParams={this.state.breadcrumbParams}
                />)
            });
    }

    renderStudyDetails() {
        return (
            <Switch>
                {routes
                    .sort((a, b) => a.navigation.order - b.navigation.order)
                    .reverse()
                    .map((route, index) => this.toRoute(index, route))}
                {routes
                    .filter(route => route.default)
                    .map((route, index) => this.toRoute(index, route))[0]
                }
            </Switch>);
    }

    toRoute(index, route) {
        return (<Route
            key={index}
            path={route.path}
            render={route.render(this.state)}/>);
    }

    setAddInvestigatorInProgress(inProgress) {
        this.setState({
            addInvestigatorInProgress: inProgress
        })
    }

    setRemoveInvestigatorInProgress(inProgress) {
        this.setState({
            removeInvestigatorInProgress: inProgress
        })
    }

    updateEdcSubjectReference(edcSubjectReference) {
        this.setState({
            breadcrumbParams: {...this.state.breadcrumbParams, edcSubjectReference: edcSubjectReference}
        })
    }

    updateEventName(event) {
        this.setState({
            breadcrumbParams: {...this.state.breadcrumbParams, eventName: event.name}
        })
    }

    updateStudyName(studyName) {
        this.setState({
            breadcrumbParams: {...this.state.breadcrumbParams, studyName: studyName}
        })
    }
}

export const mapStateToProps = function (state) {
    const {itemProvenance} = state;
    return {
        hideSidebar: Object.keys(itemProvenance).length !== 0
    }
};

export default connect(mapStateToProps)(StudyDetails);