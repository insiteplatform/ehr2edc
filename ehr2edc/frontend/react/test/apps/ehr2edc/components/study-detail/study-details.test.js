import React from 'react';
import renderer from 'react-test-renderer';
import {mapStateToProps, StudyDetails} from "../../../../../src/apps/ehr2edc/components/study-detail/study-details";
import {StaticRouter} from "react-router-dom";
import readDataFromFile from "../../../../__helper__/sample-data-file";

const studyId = "STUDY-1";

'use strict';
const studyApi = {};
const potentialInvestigatorApi = {};
const patientIdentifierSourcesApi = {};
const studyDetails = {
    studyApi: studyApi,
    potentialInvestigatorApi: potentialInvestigatorApi,
    patientIdentifierSourcesApi: patientIdentifierSourcesApi,
    studyId: studyId
};

function letStudyReturnError() {
    const error = {
        response: {
            data: {
                "issues": [{
                    "reference": null,
                    "field": null,
                    "message": "Study failed to load."
                }]
            }, headers: {}
        }
    };
    studyApi.getOne = () => Promise.reject(error);
}

function letPotentialInvestigatorReturnError() {
    const error = {
        response: {
            data: {
                "issues": [{
                    "reference": null,
                    "field": null,
                    "message": "Investigators failed to load."
                }]
            }, headers: {}
        }
    };
    potentialInvestigatorApi.getAll = () => Promise.reject(error);
}

function letWebserviceReturnStudyData(filePath) {
    const response = {data: readDataFromFile("studycontroller/" + filePath + ".json")};
    studyApi.getOne = () => Promise.resolve(response);
}

function letWebserviceReturnPotentialInvestigatorData(filePath) {
    const response = {data: readDataFromFile("userscontroller/" + filePath + ".json")};
    potentialInvestigatorApi.getAll = () => Promise.resolve(response);
}

function letWebserviceReturnPatientIdentifierSourcesApi(filePath) {
    const response = {data: readDataFromFile("patientdomaincontroller/" + filePath + ".json")};
    patientIdentifierSourcesApi.getAll = () => Promise.resolve(response);
}

function letPatientIdentifierSourcesApiReturnError() {
    const error = {};
    patientIdentifierSourcesApi.getAll = () => Promise.reject(error);
}

beforeEach(() => {
    letWebserviceReturnStudyData("getStudy-knownStudy-response-without-investigators-or-subjects");
    letWebserviceReturnPotentialInvestigatorData("unassigned-response");
    letWebserviceReturnPatientIdentifierSourcesApi("patientIdentifierSources");
});

describe("Rendering the component", () => {
    test("Loading state", async () => {
        const app = renderer.create(<StaticRouter>
            <StudyDetails {...studyDetails}/>
        </StaticRouter>);

        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
        await Promise.resolve();
    });
    test("Error state", async () => {
        letStudyReturnError();
        const app = renderer.create(<StaticRouter>
            <StudyDetails {...studyDetails}/>
        </StaticRouter>);

        await Promise.resolve();
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Error state with custom error message", async () => {
        letStudyReturnError();
        const app = renderer.create(<StaticRouter>
            <StudyDetails {...studyDetails}
                          errorMessage="Test error message"/>
        </StaticRouter>);

        await Promise.resolve();
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Valid study data json gets rendered well", async () => {
        const app = renderer.create(<StaticRouter>
            <StudyDetails {...studyDetails}/>
        </StaticRouter>);

        await Promise.resolve();
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Valid study data json gets rendered well with a custom studyPageUrl", async () => {
        const studyPageUrl = "/studyPage";
        const app = renderer.create(<StaticRouter>
            <StudyDetails {...studyDetails}
                          studyPageUrl={studyPageUrl}/>
        </StaticRouter>);

        await Promise.resolve();
        await Promise.resolve();
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Loading state when the study data is retrieved and the potential investigators aren't", async () => {
        const app = renderer.create(<StaticRouter>
            <StudyDetails {...studyDetails}/>
        </StaticRouter>);
        await Promise.resolve();
        const tree = app.toJSON();

        expect(tree).toMatchSnapshot();
    });
    test("Error state when only the potential investigators call fails", async () => {
        letPotentialInvestigatorReturnError();
        const app = renderer.create(<StaticRouter>
            <StudyDetails {...studyDetails}/>
        </StaticRouter>);

        await Promise.resolve();
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Error state when only the patient identifier sources call fails", async () => {
        letPatientIdentifierSourcesApiReturnError();
        const app = renderer.create(<StaticRouter>
            <StudyDetails {...studyDetails}
            />
        </StaticRouter>);

        await Promise.resolve();
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Firing the onInvestigatorAdd event", () => {
    let investigatorApi = {};
    const investigatorToAdd = readDataFromFile("studyinvestigatorscontroller/addInvestigators-single-investigator.json");

    beforeEach(() => {
        investigatorApi.create = (investigator) => Promise.resolve();
        studyApi.getOne = () => Promise.resolve();
    });

    test("Firing the onInvestigatorAdd event saves the correct investigator", () => {
        // Given: The component is mounted
        const app = shallow(<StudyDetails {...studyDetails}
                                          investigatorApi={investigatorApi}/>);
        app.instance().assignInvestigator = jest.fn();

        // When: Firing the onInvestigatorAdd-event
        app.state("onInvestigatorAdd")(investigatorToAdd);

        // Then: The investigator was saved
        expect(app.instance().assignInvestigator).toHaveBeenCalledTimes(1);
        expect(app.instance().assignInvestigator).toHaveBeenCalledWith(investigatorToAdd);
    });
});

describe("Firing the onRefresh event", () => {
    let investigatorApi = {};

    beforeEach(() => {
        investigatorApi = {
            create: (investigator) => Promise.resolve()
        };
    });

    test("Firing the onRefresh event refreshes the study data", () => {
        // Given: The component is mounted
        const app = shallow(<StudyDetails {...studyDetails}
                                          investigatorApi={investigatorApi}/>);
        app.instance().fetchStudy = jest.fn();

        // When: Firing the onRefresh-event
        app.state("onRefresh")();

        // Then: The study is fetched
        expect(app.instance().fetchStudy).toHaveBeenCalled();
    });
});

describe("Firing the onEdcSubjectReferenceChanged event", () => {

    test("Firing the onEdcSubjectReferenceChanged event updates edcSubjectReference in breadcrumbParams", () => {
        const app = shallow(<StudyDetails/>);
        const onEdcSubjectReferenceChanged = app.state("onEdcSubjectReferenceChanged");

        onEdcSubjectReferenceChanged("EDC_SUBJECT_REFERENCE_1");

        expect(app.state('breadcrumbParams')).toEqual({
            edcSubjectReference: "EDC_SUBJECT_REFERENCE_1",
            eventName: "",
            studyName: ""
        })
    });
});

describe("Firing the onEventChanged event", () => {

    test("Firing the onEventChanged event updates event name in breadcrumbParams", () => {
        const app = shallow(<StudyDetails/>);
        const onEventChanged = app.state("onEventChanged");

        onEventChanged({name: "EVENT_NAME_1"});

        expect(app.state('breadcrumbParams')).toEqual({
            edcSubjectReference: "",
            eventName: "EVENT_NAME_1",
            studyName: ""
        })
    });
});

describe("mapStateToProps", () => {
    test("When itemProvenance is empty, hideSidebar is false", () => {
        const state = {itemProvenance: {}};

        expect(mapStateToProps(state)).toEqual({hideSidebar: false});
    });
    test("When itemProvenance is present, hideSidebar is true", () => {
        const state = {itemProvenance: {key: "value"}};

        expect(mapStateToProps(state)).toEqual({hideSidebar: true});
    });
});