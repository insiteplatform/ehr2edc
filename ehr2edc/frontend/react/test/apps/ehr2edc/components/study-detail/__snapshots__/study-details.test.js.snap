// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the component Error state 1`] = `
Array [
  <div
    className="sidebar grid-block vertical collapsible shrink sidebar-nav border-right"
  >
    <div
      className="grid-content padding shrink border-bottom"
    >
      <h4 />
    </div>
    <div
      className="grid-block vertical"
    >
      <ul
        className="side-bar-navigation"
      >
        <li>
          <a
            href="/STUDY-1/"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabOverview"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabOverview"
            >
              Overview
            </label>
          </a>
        </li>
        <li>
          <a
            href="/STUDY-1/investigators"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabInvestigators"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabInvestigators"
            >
              Investigators
            </label>
          </a>
        </li>
        <li>
          <div>
            <input
              className="radioTab disabled"
              id="radioTabParticipants"
              name="radioTab"
              type="radio"
            />
            <label
              htmlFor="radioTabParticipants"
            >
              Participants
            </label>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="toggle border-top"
      onClick={[Function]}
    >
      <i
        className="fa fa-angle-double-left"
        title="Hide"
      />
      <span>
        Hide
      </span>
    </div>
  </div>,
  <div
    className="grid-block vertical"
    id="study-details-main-pane"
  >
    <ul
      className="bread-crumbs shrink grid-block border-bottom"
      id="page-bread-crumbs"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          EHR2EDC Studies
        </a>
      </li>
    </ul>
    <div
      className="grid-block"
      id="study-details-content-pane"
    >
      <div
        className="grid-block vertical"
      >
        <div
          className="grid-block search-results-loading"
        >
          <span
            className="search-results-loading-icon"
          >
            <i
              className="fa fa-circle-o-notch fa-spin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Rendering the component Error state when only the patient identifier sources call fails 1`] = `
Array [
  <div
    className="sidebar grid-block vertical collapsible shrink sidebar-nav border-right"
  >
    <div
      className="grid-content padding shrink border-bottom"
    >
      <h4 />
    </div>
    <div
      className="grid-block vertical"
    >
      <ul
        className="side-bar-navigation"
      >
        <li>
          <a
            href="/STUDY-1/"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabOverview"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabOverview"
            >
              Overview
            </label>
          </a>
        </li>
        <li>
          <a
            href="/STUDY-1/investigators"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabInvestigators"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabInvestigators"
            >
              Investigators
            </label>
          </a>
        </li>
        <li>
          <div>
            <input
              className="radioTab disabled"
              id="radioTabParticipants"
              name="radioTab"
              type="radio"
            />
            <label
              htmlFor="radioTabParticipants"
            >
              Participants
            </label>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="toggle border-top"
      onClick={[Function]}
    >
      <i
        className="fa fa-angle-double-left"
        title="Hide"
      />
      <span>
        Hide
      </span>
    </div>
  </div>,
  <div
    className="grid-block vertical"
    id="study-details-main-pane"
  >
    <ul
      className="bread-crumbs shrink grid-block border-bottom"
      id="page-bread-crumbs"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          EHR2EDC Studies
        </a>
      </li>
    </ul>
    <div
      className="grid-block"
      id="study-details-content-pane"
    >
      <div
        className="grid-block vertical"
      >
        <div
          className="grid-block search-results-loading"
        >
          <span
            className="search-results-loading-icon"
          >
            <i
              className="fa fa-circle-o-notch fa-spin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Rendering the component Error state when only the potential investigators call fails 1`] = `
Array [
  <div
    className="sidebar grid-block vertical collapsible shrink sidebar-nav border-right"
  >
    <div
      className="grid-content padding shrink border-bottom"
    >
      <h4 />
    </div>
    <div
      className="grid-block vertical"
    >
      <ul
        className="side-bar-navigation"
      >
        <li>
          <a
            href="/STUDY-1/"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabOverview"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabOverview"
            >
              Overview
            </label>
          </a>
        </li>
        <li>
          <a
            href="/STUDY-1/investigators"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabInvestigators"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabInvestigators"
            >
              Investigators
            </label>
          </a>
        </li>
        <li>
          <div>
            <input
              className="radioTab disabled"
              id="radioTabParticipants"
              name="radioTab"
              type="radio"
            />
            <label
              htmlFor="radioTabParticipants"
            >
              Participants
            </label>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="toggle border-top"
      onClick={[Function]}
    >
      <i
        className="fa fa-angle-double-left"
        title="Hide"
      />
      <span>
        Hide
      </span>
    </div>
  </div>,
  <div
    className="grid-block vertical"
    id="study-details-main-pane"
  >
    <ul
      className="bread-crumbs shrink grid-block border-bottom"
      id="page-bread-crumbs"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          EHR2EDC Studies
        </a>
      </li>
    </ul>
    <div
      className="grid-block"
      id="study-details-content-pane"
    >
      <div
        className="grid-block vertical"
      >
        <div
          className="grid-block search-results-loading"
        >
          <span
            className="search-results-loading-icon"
          >
            <i
              className="fa fa-circle-o-notch fa-spin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Rendering the component Error state with custom error message 1`] = `
Array [
  <div
    className="sidebar grid-block vertical collapsible shrink sidebar-nav border-right"
  >
    <div
      className="grid-content padding shrink border-bottom"
    >
      <h4 />
    </div>
    <div
      className="grid-block vertical"
    >
      <ul
        className="side-bar-navigation"
      >
        <li>
          <a
            href="/STUDY-1/"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabOverview"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabOverview"
            >
              Overview
            </label>
          </a>
        </li>
        <li>
          <a
            href="/STUDY-1/investigators"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabInvestigators"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabInvestigators"
            >
              Investigators
            </label>
          </a>
        </li>
        <li>
          <div>
            <input
              className="radioTab disabled"
              id="radioTabParticipants"
              name="radioTab"
              type="radio"
            />
            <label
              htmlFor="radioTabParticipants"
            >
              Participants
            </label>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="toggle border-top"
      onClick={[Function]}
    >
      <i
        className="fa fa-angle-double-left"
        title="Hide"
      />
      <span>
        Hide
      </span>
    </div>
  </div>,
  <div
    className="grid-block vertical"
    id="study-details-main-pane"
  >
    <ul
      className="bread-crumbs shrink grid-block border-bottom"
      id="page-bread-crumbs"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          EHR2EDC Studies
        </a>
      </li>
    </ul>
    <div
      className="grid-block"
      id="study-details-content-pane"
    >
      <div
        className="grid-block vertical"
      >
        <div
          className="grid-block search-results-loading"
        >
          <span
            className="search-results-loading-icon"
          >
            <i
              className="fa fa-circle-o-notch fa-spin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Rendering the component Loading state 1`] = `
Array [
  <div
    className="sidebar grid-block vertical collapsible shrink sidebar-nav border-right"
  >
    <div
      className="grid-content padding shrink border-bottom"
    >
      <h4 />
    </div>
    <div
      className="grid-block vertical"
    >
      <ul
        className="side-bar-navigation"
      >
        <li>
          <a
            href="/STUDY-1/"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabOverview"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabOverview"
            >
              Overview
            </label>
          </a>
        </li>
        <li>
          <a
            href="/STUDY-1/investigators"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabInvestigators"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabInvestigators"
            >
              Investigators
            </label>
          </a>
        </li>
        <li>
          <div>
            <input
              className="radioTab disabled"
              id="radioTabParticipants"
              name="radioTab"
              type="radio"
            />
            <label
              htmlFor="radioTabParticipants"
            >
              Participants
            </label>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="toggle border-top"
      onClick={[Function]}
    >
      <i
        className="fa fa-angle-double-left"
        title="Hide"
      />
      <span>
        Hide
      </span>
    </div>
  </div>,
  <div
    className="grid-block vertical"
    id="study-details-main-pane"
  >
    <ul
      className="bread-crumbs shrink grid-block border-bottom"
      id="page-bread-crumbs"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          EHR2EDC Studies
        </a>
      </li>
    </ul>
    <div
      className="grid-block"
      id="study-details-content-pane"
    >
      <div
        className="grid-block vertical"
      >
        <div
          className="grid-block search-results-loading"
        >
          <span
            className="search-results-loading-icon"
          >
            <i
              className="fa fa-circle-o-notch fa-spin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Rendering the component Loading state when the study data is retrieved and the potential investigators aren't 1`] = `
Array [
  <div
    className="sidebar grid-block vertical collapsible shrink sidebar-nav border-right"
  >
    <div
      className="grid-content padding shrink border-bottom"
    >
      <h4 />
    </div>
    <div
      className="grid-block vertical"
    >
      <ul
        className="side-bar-navigation"
      >
        <li>
          <a
            href="/STUDY-1/"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabOverview"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabOverview"
            >
              Overview
            </label>
          </a>
        </li>
        <li>
          <a
            href="/STUDY-1/investigators"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabInvestigators"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabInvestigators"
            >
              Investigators
            </label>
          </a>
        </li>
        <li>
          <div>
            <input
              className="radioTab disabled"
              id="radioTabParticipants"
              name="radioTab"
              type="radio"
            />
            <label
              htmlFor="radioTabParticipants"
            >
              Participants
            </label>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="toggle border-top"
      onClick={[Function]}
    >
      <i
        className="fa fa-angle-double-left"
        title="Hide"
      />
      <span>
        Hide
      </span>
    </div>
  </div>,
  <div
    className="grid-block vertical"
    id="study-details-main-pane"
  >
    <ul
      className="bread-crumbs shrink grid-block border-bottom"
      id="page-bread-crumbs"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          EHR2EDC Studies
        </a>
      </li>
    </ul>
    <div
      className="grid-block"
      id="study-details-content-pane"
    >
      <div
        className="grid-block vertical"
      >
        <div
          className="grid-block search-results-loading"
        >
          <span
            className="search-results-loading-icon"
          >
            <i
              className="fa fa-circle-o-notch fa-spin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Rendering the component Valid study data json gets rendered well 1`] = `
Array [
  <div
    className="sidebar grid-block vertical collapsible shrink sidebar-nav border-right"
  >
    <div
      className="grid-content padding shrink border-bottom"
    >
      <h4 />
    </div>
    <div
      className="grid-block vertical"
    >
      <ul
        className="side-bar-navigation"
      >
        <li>
          <a
            href="/STUDY-1/"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabOverview"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabOverview"
            >
              Overview
            </label>
          </a>
        </li>
        <li>
          <a
            href="/STUDY-1/investigators"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabInvestigators"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabInvestigators"
            >
              Investigators
            </label>
          </a>
        </li>
        <li>
          <div>
            <input
              className="radioTab disabled"
              id="radioTabParticipants"
              name="radioTab"
              type="radio"
            />
            <label
              htmlFor="radioTabParticipants"
            >
              Participants
            </label>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="toggle border-top"
      onClick={[Function]}
    >
      <i
        className="fa fa-angle-double-left"
        title="Hide"
      />
      <span>
        Hide
      </span>
    </div>
  </div>,
  <div
    className="grid-block vertical"
    id="study-details-main-pane"
  >
    <ul
      className="bread-crumbs shrink grid-block border-bottom"
      id="page-bread-crumbs"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          EHR2EDC Studies
        </a>
      </li>
    </ul>
    <div
      className="grid-block"
      id="study-details-content-pane"
    >
      <div
        className="grid-block vertical"
      >
        <div
          className="grid-block search-results-loading"
        >
          <span
            className="search-results-loading-icon"
          >
            <i
              className="fa fa-circle-o-notch fa-spin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Rendering the component Valid study data json gets rendered well with a custom studyPageUrl 1`] = `
Array [
  <div
    className="sidebar grid-block vertical collapsible shrink sidebar-nav border-right"
  >
    <div
      className="grid-content padding shrink border-bottom"
    >
      <h4 />
    </div>
    <div
      className="grid-block vertical"
    >
      <ul
        className="side-bar-navigation"
      >
        <li>
          <a
            href="/STUDY-1/"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabOverview"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabOverview"
            >
              Overview
            </label>
          </a>
        </li>
        <li>
          <a
            href="/STUDY-1/investigators"
            onClick={[Function]}
          >
            <input
              className="radioTab"
              id="radioTabInvestigators"
              name="radioTab"
              type="radio"
              value="true"
            />
            <label
              htmlFor="radioTabInvestigators"
            >
              Investigators
            </label>
          </a>
        </li>
        <li>
          <div>
            <input
              className="radioTab disabled"
              id="radioTabParticipants"
              name="radioTab"
              type="radio"
            />
            <label
              htmlFor="radioTabParticipants"
            >
              Participants
            </label>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="toggle border-top"
      onClick={[Function]}
    >
      <i
        className="fa fa-angle-double-left"
        title="Hide"
      />
      <span>
        Hide
      </span>
    </div>
  </div>,
  <div
    className="grid-block vertical"
    id="study-details-main-pane"
  >
    <ul
      className="bread-crumbs shrink grid-block border-bottom"
      id="page-bread-crumbs"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          EHR2EDC Studies
        </a>
      </li>
    </ul>
    <div
      className="grid-block"
      id="study-details-content-pane"
    >
      <div
        className="grid-block vertical"
      >
        <div
          className="grid-block search-results-loading"
        >
          <span
            className="search-results-loading-icon"
          >
            <i
              className="fa fa-circle-o-notch fa-spin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;
